src/reputation/tasks.py

```diff
@@ -18,17 +20,76 @@
 from mailing_list.lib import base_email_context
 from paper.models import Figure, Paper
 from reputation.distributor import RewardDistributor
-from reputation.models import Contribution, DistributionAmount
-from researchhub.celery import QUEUE_CONTRIBUTIONS, QUEUE_PURCHASES, app
+from reputation.models import Bounty, Contribution, DistributionAmount
+from researchhub.celery import (
+    QUEUE_BOUNTY_NOTIFICATIONS,
+    QUEUE_CONTRIBUTIONS,
+    QUEUE_PURCHASES,
+    app,
+)
 from researchhub.settings import BASE_FRONTEND_URL
-from researchhub_document.models import ResearchhubUnifiedDocument
+from researchhub_document.models import ResearchhubPost, ResearchhubUnifiedDocument
 from summary.models import Summary
 from utils.message import send_email_message
 from utils.sentry import log_info
 
 DEFAULT_REWARD = 1000000
 
 
+def build_notification_context(message, subject, frontend_view_link):
+    return {
+        **base_email_context,
+        "frontend_view_link": frontend_view_link,
+        "message": message,
+        "subject": subject,
+    }
+
+
+@periodic_task(
+    run_every=crontab(minute="0", hour="10"),
+    priority=3,
+    queue=QUEUE_BOUNTY_NOTIFICATIONS,
+)
+def send_bounty_notifications():
+    twentynine_days_ago = start_date = datetime.datetime.now() - datetime.timedelta(29)
+    bounties = Bounty.objects.filter(
+        status=Bounty.OPEN,
+        created_date__date=twentynine_days_ago,
```

Comment:
Is it possible to add a condition that examines if the user received a notification already? Reason being is, if we tweak the filtering logic, it could result in user getting more than one notification or not getting any at all.